src/ImpurityDensities/PtypePNJunctionImpurityDensity.jl

```diff
@@ -0,0 +1,55 @@
+"""
+    struct ThermalDiffusionLithiumDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+A PN junction impurity model based on lithium thermal diffusion and constant bulk impurity density.
+The surface lithium density is saturated.
+
+ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+ 
+## Fields
+* `lithium_annealing_temperature::T`: lithium annealing temperature when the lithium is diffused into the crystal. The default value is 623 K.
+* `lithium_annealing_time::T`: lithium annealing time. The default value is 18 minutes.
+* `surface_imp_model::ThermalDiffusionLithiumDensity{T,G}`: the density profile of lithium (n-type). This will be constructed with above parameters.
+* `bulk_imp_model::AbstractImpurityDensity{T}`: the density of the p-type impurity.
+"""
+
+struct PtypePNJunctionImpurityDensity{T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}} <: AbstractImpurityDensity{T}
+	lithium_annealing_temperature::T
+	lithium_annealing_time::T
+    contact_with_lithium_doped::G
+    surface_imp_model::ThermalDiffusionLithiumDensity{T,G}
+    bulk_imp_model::AbstractImpurityDensity{T}
+end
+function PtypePNJunctionImpurityDensity{T,G}(
+    lithium_annealing_temperature::T,
+    lithium_annealing_time::T,
+    contact_with_lithium_doped::G,
+    bulk_imp_model::AbstractImpurityDensity{T}
+    ) where {T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}}
+    surface_imp_model = ThermalDiffusionLithiumDensity{T,G}(lithium_annealing_temperature, lithium_annealing_time, contact_with_lithium_doped)
+    PtypePNJunctionImpurityDensity{T,G}(lithium_annealing_temperature,lithium_annealing_time, contact_with_lithium_doped, surface_imp_model, bulk_imp_model)
+end
+function PtypePNJunctionImpurityDensity{T,G}(
+    lithium_annealing_temperature::T,
+    lithium_annealing_time::T,
+    contact_with_lithium_doped::G,
+    bulk_imp_model::AbstractImpurityDensity{T},
+    distance_to_contact::Function
+    ) where {T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}}
+    surface_imp_model = ThermalDiffusionLithiumDensity{T,G}(lithium_annealing_temperature, lithium_annealing_time, contact_with_lithium_doped, distance_to_contact=distance_to_contact)
+    PtypePNJunctionImpurityDensity{T,G}(lithium_annealing_temperature,lithium_annealing_time, contact_with_lithium_doped, surface_imp_model, bulk_imp_model)
+end
```

Comment:
This can become one function if you pass `distance_to_contact` a default function.

```suggestion
function PtypePNJunctionImpurityDensity{T,G}(
    lithium_annealing_temperature::T,
    lithium_annealing_time::T,
    contact_with_lithium_doped::G,
    bulk_imp_model::AbstractImpurityDensity{T},
    distance_to_contact::Function = pt::AbstractCoordinatePoint{T} -> ConstructiveSolidGeometry.distance_to_surface(pt, contact_with_lithium_doped)
    ) where {T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}}
    surface_imp_model = ThermalDiffusionLithiumDensity{T,G}(lithium_annealing_temperature, lithium_annealing_time, contact_with_lithium_doped, distance_to_contact=distance_to_contact)
    PtypePNJunctionImpurityDensity{T,G}(lithium_annealing_temperature,lithium_annealing_time, contact_with_lithium_doped, surface_imp_model, bulk_imp_model)
end
```